dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -23,25 +23,45 @@ def self.truncate_table
   end
 
   def self.extract_email_preferences
-    query = extract_from_source_query('email_preferences', ['opt_in'], 'email')
+    query = get_extraction_query("#{CDO.dashboard_db_name}.email_preferences", false, ['opt_in'], 'email')
```

Comment:
Non-blocking: what if we change the parameter order by moving `source_is_query` to the end (closer to `source_name`) and set its default value to `false`? This line will be shorter and read more smoothly: _get extraction query for \<a table\> for \<several columns\>_.

```suggestion
    query = get_extraction_query("#{CDO.dashboard_db_name}.email_preferences",  'email', ['opt_in'])
```

Reply:
Fixed, much better!